Remove "Changelog" section of lmms.spec.in

Wallacoloo opened the pull request:
The change log can be accessed through the repository history, which is far more fool-proof than storing it as data in the actual repository.

curlymorphic commented:
I agree with this, The removed data is also 7 years out of date.

tresf commented:
This may have been introduced to offer some information to the RPM package as these packages often store supplemental information for a package manager.

Although I'm in favor of removing this, I think we need to rethink our package management techniques in general (for example, @falkTX has some VST-specific packaging steps he uses for KXStudio that we may be able to incorporate upstream).  I would even go as far as to say we should have a way of reading the github release information and injecting it at packaging time, which would be a project in itself.

It would be nice if we had someone with Fedora/RedHat packaging experience to help out with this, but I wanted to mention the potential value of this section before we remove it completely.

Should we consider keeping the section but replacing it with some `lorem ipsum` boilerplate text for now?

-Tres

Wallacoloo commented:
> This may have been introduced to offer some information to the RPM package as these packages often store supplemental information for a package manager.

I see, so then it may be worth not wiping it altogether.

> Should we consider keeping the section but replacing it with some lorem ipsum boilerplate text for now?

No. I'd say either leave it as-is or (confirm its use and- ) fix it. I don't see any advantage of placeholder text over what's currently in there.

tresf commented:
Hmm... How about a compromise... Can we put a "please see release notes" and hyperlink to the GitHub release info for that version?

tresf commented:
Pseudo code:

```
Please see release notes:

https://github.com/LMMS/lmms/releases/tag/v${LMMS_VERSION}

Wallacoloo commented:
@tresf: that seems reasonable. I'll update the PR for that in a few hours
once I get access to a computer, unless @curlymorphic has anything to say
about your suggestion.

tresf commented:
Thanks.  Off topic but I don't think we'll hit merge conflicts with this change (against #2210) since I used the `git mv` command, but we'll see. :+1:

Wallacoloo commented:
Updated. I also fixed the version tag (0.3.0) to use ${LMMS_VERSION} as well. This is the first time that the ${CMAKE_VARIABLE} syntax has been used in that file though, and I have absolutely no idea where the generated lmms.spec file is placed during the build (or even how to generate it), so I could not test the changes.

tresf commented:
Thanks.  The `.in` extension suggests it is configured by CMake, but I do not believe any targets have been setup to process this.

Moving it to the data directory could put the task there (e.g. /data/CMakeLists.text) if we'd like but so long as no one is using it, the downstream effect is nothing.

If someone is using it (such as for RPM packaging) they're likely to complain, but the info is so drastically outdated, they're already used to hand-editing it, so there is zero risk IMO.

tresf commented:
Merging.  We can change `CMakeList.txt` to generate the proper `.spec` file at another time when we've determined the actual usefulness of such a file.
